Cython/Compiler/FusedNode.py

```diff
@@ -686,31 +715,61 @@ def __pyx_fused_cpdef(signatures, args, kwargs, defaults):
             self._buffer_declarations(pyx_code, decl_code, all_buffer_types, pythran_types)
             env.use_utility_code(Code.UtilityCode.load_cached("Import", "ImportExport.c"))
             env.use_utility_code(Code.UtilityCode.load_cached("ImportNumPyArray", "ImportExport.c"))
+        
+        context.update(
+            global_sigindex_name = fused_cpdef_globalindex,
+            func_sigindex_key = env.mangle(fused_cpdef_globalindex_prefix, orig_py_func.entry.cname)
+        )
 
         pyx_code.put_chunk(
             u"""
-                candidates = []
-                for sig in <dict>signatures:
-                    match_found = False
-                    src_sig = sig.strip('()').split('|')
-                    for i in range(len(dest_sig)):
-                        dst_type = dest_sig[i]
-                        if dst_type is not None:
-                            if src_sig[i] == dst_type:
-                                match_found = True
-                            else:
-                                match_found = False
-                                break
+                global {{global_sigindex_name}}
+                if '{{global_sigindex_name}}' not in <dict>globals():
```

Comment:
There may be a good reason why you didn't do this, but why not just do:

```
cdef dict {{global_sigindex_name}} = {}
```

outside the function scope? That way you know it's always set up, and you don't have to do the `not in globals()` check.

Obviously this only applies if you keep it in `globals()`

Reply:
Looking at it a bit more - I think this solves (most of) the issue with leaking the variable to the user. Module level `cdef` variables are publicly accessible outside of the Cython module. You could probably still access the name inside the Cython module, but that's less of a concern since it doesn't appear in `dir(module)`

Reply:
There is a reason, but not a good one. Mostly it's just because I didn't know how to do it, and putting it in the module's dynamic `__dict__` (I.E. accessible from CPython) was useful for inspecting it.

It would have to be part of the boilerplate for the module itself, correct? And ideally, it should maybe be either absent or declared as a null value unless the module includes `fused` `cpdef` functions.

Reply:
There is a reason, but not a good one. Mostly it's just because I didn't know how to do it, and putting it in the module's dynamic `__dict__` (I.E. accessible from CPython) was useful for inspecting it.

It would have to be part of the boilerplate for the module itself, correct? And ideally, it should maybe be either absent or declared as a null value unless the module includes `fused` `cpdef` functions.

Reply:
I'm going to try putting it in `ParseTreeTransforms.py`, following the example of `__pyx_unpickle_`.

Reply:
I don't think moving it to ParseTreeTransforms will help, but it'd definitely be a good idea to look at the code in ParseTreeTransforms and see there's anything you can copy.

Reply:
It's kind of strange. Appending the generated definition to `<list>ParseTreeTransforms.AnalyseDeclarationsTransform().extra_module_declarations` with a hook from `._handle_def()` works (as an entry in the module's `__dict__`) when I do a normal assignment in it with `__pyx_fusedcpdef_sigindices = {}`, but doing `cdef __pyx_fusedcpdef_sigindices in the same place` neither helps nor crashes it.